overlord/snapstate/snapstate.go

```diff
@@ -154,6 +154,20 @@ func (ins installSnapInfo) SnapSetupForUpdate(st *state.State, params updatePara
 		return nil, nil, err
 	}
 
+	var registries [][2]string
+	for _, plug := range ins.Plugs {
+		if plug.Interface != "registry" {
+			continue
+		}
+
+		account, registry, _, err := snap.RegistryPlugAttrs(plug)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		registries = append(registries, [2]string{account, registry})
+	}
```

Comment:
This is a bit of duplicated code but I didn't refactor given that it's likely to disappear once this is merged with the ongoing snapstate refactor https://github.com/snapcore/snapd/pull/14146